pyntrinio/pyntrinio.py

```diff
@@ -189,15 +189,65 @@ def gather_stock_time_series(api_key, ticker, start_date, end_date, output_forma
   Example
   -----------
   >>> gather_stock_time_series(api_key, 'AAPL')
-  {'ticker':['AAPL'], 'adj_close':[300], 'date':[datetime.date(2020, 1, 2)]}
   """
+    # ensure the type of the ticker is a string
+    if type(ticker) != str:
+        print("Invalid Input: ticker has to be a string, e.g. 'APPL'")
+        return
+    
+    try:
+        # change dates to datetime objects
+        if start_date is not None:
+            start_date = datetime.strptime(start_date, '%Y-%m-%d').date()
+        if end_date is not None:
+            end_date = datetime.strptime(end_date, '%Y-%m-%d').date()
+    except:
+        print("Invalid Date format - please input the date as a string with format %Y-%m-%d")
+        return
+    
+    if start_date >= end_date:
+        print("Invalid Input: end_date is earlier than start_date")
+        return
```

Comment:
Only when both dates are not `None`, that they can be compared. Otherwise, it will give errors.

Reply:
updated